components/SignUp.js

```diff
@@ -1,81 +1,160 @@
-        return alert("An account with that email is already in use!");
-      } else {
-        return alert("Error creating user!");
-      }
-    }
-  }
+  // Handles input changes on submit.
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormValues({ ...formValues, [name]: value });
+  };
 
-  // Password strength check.
-  function passwordStrength() {
-    const passwordInput = document.getElementById("password");
-    const StrengthIndicator =
-      document.querySelector("#strength-indicator");
+  // Handles submit.
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setFormErrors(validate(formValues));
+    setIsSubmit(true);
+  };
 
-    if (passwordInput.value.match(/^(?=.*[a-z]).{6,20}$/)) {
-      StrengthIndicator.textContent = "Password strength: Weak!";
+  // Validate form. Displays error messages. If no errors on submit true, call postData().
+  useEffect(() => {
+    if (Object.keys(formErrors).length === 0 && isSubmit) {
+      postData(formValues);
+    }
+  }, [formErrors]);
+  const validate = (values) => {
+    const errors = {};
+    const regex = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
+    if (!values.name) {
+      errors.name = "Du måste ange ett fullständigt namn!";
+    }
+    if (!values.phone) {
+      errors.phone = "Du måste ange ett telefonnummer!";
+    } else if (values.phone.length < 10) {
+      errors.phone = "Ett telefonnummer måste vara minst 10 siffror!";
     }
-    if (passwordInput.value.match(/^(?=.*[a-z])(?=.*[A-Z]).{6,20}$/)) {
-      StrengthIndicator.textContent =
-        "Password strength: Medium!";
+    if (!values.email) {
+      errors.email = "Du måste ange en e-postadress!";
+    } else if (!regex.test(values.email)) {
+      errors.email = "Det här är inte en giltig e-postadress!";
     }
-    if (passwordInput.value.match(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,20}$/)) {
-      StrengthIndicator.textContent = "Password strength: Strong!";
+    if (!values.password) {
+      errors.password = "Du måste ange ett lösenord med minst 4 tecken!";
+    } else if (values.password.length < 4) {
+      errors.password = "Lösenordet måste vara minst 4 tecken långt!";
+    } else if (values.password.length > 20) {
+      errors.password = "Lösenordet får vara max 20 tecken långt!";
     }
-    if (
-      passwordInput.value.match(
-        /^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{6,20}$/
-      )
-    ) {
-      StrengthIndicator.textContent = "Password strength: Very strong!";
+    if (!values.verifyPassword) {
+      errors.verifyPassword = "Du måste ange samma lösenord igen!";
+    } else if (values.verifyPassword !== values.password) {
+      errors.verifyPassword = "Lösenordet överensstämmer inte!";
     }
+    return errors;
   };
 
-  // Render the signUp component.
+  // Check password strength.
+  function strengthCheck() {
+    if (formValues.password < 4) {
+      setpasswordStrength("Lösenordets styrka: Inte tillräcklig");
```

Comment:
This is mixing data (the strength of the password) and presentation (the Swedish phrase used to communicate the strength to the user). The data would be better represented by something more concise, e.g. a number scale (0-5), a set of constants (`TOO_WEAK`, `WEAK`, `AVERAGE` etc) or if you had been using typescript, an `enum`.

That makes it easier to check in different parts of the code if the password is strong enough, without having to know the exact Swedish phrase used. It also makes it possible to change the presentation (i.e. the Swedish wording) without changing any logic.